python/lsst/source/injection/utils/_make_injection_pipeline.py

```diff
@@ -166,6 +166,21 @@ def make_injection_pipeline(
             ", ".join(sorted(not_found_tasks)),
         )
 
+    # Check for any empty subsets and remove them.
+    removed_subsets = set()
+    for subset_label, subset_tasks in pipeline.subsets.items():
+        if not subset_tasks:
+            removed_subsets.add(subset_label)
+            pipeline.removeLabeledSubset(subset_label)
+    if len(removed_subsets) > 0:
```

Comment:
The `len` calls in some many places jumps out to me, consider:
```python
if (size := len(removed_subsets)) > 0:
    grammar = "subset" if size == 1 else subsets
    logger.warning(...,size,...)
```

You don't need to make the change, just something to consider